catroid/src/org/catrobat/catroid/content/bricks/WhenTouchDownBrick.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2016 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.bricks;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.drawable.Drawable;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.CompoundButton;
+import android.widget.TextView;
+
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.WhenTouchDownScript;
+
+import java.util.List;
+
+public class WhenTouchDownBrick extends ScriptBrick {
+	protected WhenTouchDownScript whenTouchDownScript;
+	private transient View prototypeView;
+	private static final long serialVersionUID = 1L;
+	private transient AdapterView<?> adapterView;
+
+	public WhenTouchDownBrick() {
+		this.whenTouchDownScript = new WhenTouchDownScript();
+	}
+
+	public WhenTouchDownBrick(WhenTouchDownScript script) {
+		this.whenTouchDownScript = script;
+	}
+
+	@Override
+	public Brick copyBrickForSprite(Sprite sprite) {
+		WhenTouchDownBrick copyBrick = (WhenTouchDownBrick) clone();
+
+		copyBrick.whenTouchDownScript = whenTouchDownScript;
```

Comment:
I am not really sure, but don't you need to clone the script here ? I think if you do it that way, and change something in the script in the original Sprite, it will also change in the sprite you copied to, since it is the same script object.